File: gpt-4/template-context-v2/data_construct_delete_clause.c
#include "acc_testsuite.h"
#ifndef T1
//T1:data construct delete clause,V:2.5-3.0
int test1(){
    int err = 0;
    srand(SEED);
    int array[100];
    int i;
    int * dev_ptr;

    for(i = 0; i < 100; i++) {
        array[i] = i;
    }

    #pragma acc enter data create(array[:100])
    dev_ptr = acc_deviceptr(array);

    #pragma acc exit data delete(array[:50])
    
    if((int*) acc_deviceptr(array) != dev_ptr) {
        err = 1;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: bad-tests-v2/data_construct_delete_clause.f90
fortran
#ifndef T1
!T1:data construct delete clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER :: var1, var2
        INTEGER :: ptr1, ptr2
        INTEGER :: dynamic_ref_counter1, dynamic_ref_counter2
        INTEGER :: structured_ref_counter1, structured_ref_counter2
        
        ! Initialize variables
        var1 = 1
        var2 = 2
        ptr1 = 1
        ptr2 = 2
        dynamic_ref_counter1 = 0
        dynamic_ref_counter2 = 0
        structured_ref_counter1 = 0
        structured_ref_counter2 = 0
        
        ! Test data construct delete clause
        !$acc data copy(var1, var2) delete(ptr1, ptr2)
        !$acc end data
        
        ! Check if delete clause worked correctly
        IF (dynamic_ref_counter1 .eq. 0 .and. dynamic_ref_counter2 .eq. 0) THEN
          ! Delete action should have been performed
          errors = errors + 1
        END IF
        
        ! Check if detach action was performed correctly
        IF (ptr1 .eq. 1 .and. ptr2 .eq. 2) THEN
          ! Detach action should have been performed
          errors = errors + 1
        END IF
        
        ! Check if present decrement action was performed correctly
        IF (structured_ref_counter1 .eq. 0 .and. structured_ref_counter2 .eq. 0) THEN
          ! Present decrement action should have been performed
          errors = errors + 1
        END IF
        
        ! Return errors
        test1 = errors .eq. 0
      END
#endif
      
      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM